crates/solver/src/solver/single_order_solver.rs

```diff
@@ -56,6 +56,15 @@ impl<I: SingleOrderSolving> Solver for SingleOrderSolver<I> {
         let settle = async {
             while let Some(order) = orders.pop_front() {
                 match self.inner.try_settle_order(order.clone(), &auction).await {
+                    Ok(None) => {
+                        let name = self.inner.name();
+                        self.metrics.single_order_solver_failed(name);
```

Comment:
Won't we keep on getting errors this way since we are incrementing the "solver_failed" metric?

I would think that the branch:
```
Ok(settlement) => { ... }
```

is already sufficient (i.e. no new settlement found and "successful" solve - as we usually treat no liquidity as a non-error in the Baseline solver for example).

Reply:
Yeah, I was very unsure about this. I see your argument for resolving it as a success.
I introduced this match-branch mostly for the debug message, so that at least on the debug msg level we can very quickly see whether a solution was found or not.

But I checked:
One can still find this information relatively quickly in the logs by investigating the 1inch logs like this:
https://logs.gnosisdev.com/goto/b64cc120-e24f-11ec-ac8d-6902ab016f37

So you convinced me :)